### PR TITLE
12712 GMaps in builder

### LINKED ISSUE #12712: Maps with google basemap open in editor
Currently, editor maps with google basemaps are never migrated to builder: https://github.com/CartoDB/cartodb/blob/master/app/models/carto/visualization.rb#L534

This is a remnant  from when Builder did not support google basemaps, but that is no longer an issue. I think we can remove that check, and allow maps with google basemaps to be opened in Builder. Do you agree @noguerol?

cc @xavijam @juanignaciosl @ernesmb

### PR DESCRIPTION
**Pull request by amartinj**
Fixes #12712

### CODE REVIEW

**Comment by amartinj** [comment]
retest this, please

**Comment by amartinj** [comment]
retest this, please

**Comment by amartinj** [comment]
please retest this

**Comment by amartinj** [comment]
retest this, please

**Comment by amartinj** [comment]
retest this, please

**Comment by amartinj** [comment]
retest this, please

**Comment by javitonino** [comment]
🔴 in acceptance. This fails when the mapcap has to be created. Basically, the mapcap is created before applying the layer fix, so the mapcap is incorrect and causes problems. Mapcapping should be done after fixing the layers.

File: lib/carto/visualization_migrator.rb [hunk 1]
```diff
@@ -32,5 +33,15 @@ def layer_selector_migration(vis)
 
       vis.overlays.select { |o| o.type == 'layer_selector' }.each(&:destroy)
     end
+
+    def google_basemap_migration(vis)
+      # Editor gmaps were saved with base_type attribute in options, but, in builder, baseType is expected
+      vis = Carto::Visualization.find(vis.id)
```
Comment (juanignaciosl) [inline lib/carto/visualization_migrator.rb]: As we can probably receive a `Carto::Visualization`, consider doing the `find` only if `vis` class is not `Carto::Visualization`. If we can be sure that `vis` is always a `Visualization::Member`, ignore this comment.

File: lib/carto/visualization_migrator.rb [hunk 2]
```diff
@@ -32,5 +33,15 @@ def layer_selector_migration(vis)
...
+      vis.layers.select { |l| l.gmapsbase? && l.options.has_key?(:base_type) }.each do |l|
+        l.options[:baseType] = l.options[:base_type]
```
Comment (juanignaciosl) [inline lib/carto/visualization_migrator.rb]: I think that you can merge these lines into one:

```ruby
l.options[:baseType] = l.options.delete(:base_type)
```

File: lib/carto/visualization_migrator.rb (hunk 1 above)
Comment (amartinj) [reply lib/carto/visualization_migrator.rb]: I'll change it but i'd say i'd rather have cleaner code than saving one query by an indexed id. Tradeofs :P

**Comment by amartinj** [comment]
retest this, please

**Comment by javitonino** [comment]
cc @ernesmb to be notified on merge

**Pull request merged**